modules/core/src/main/java/org/apache/ignite/internal/processors/security/IgniteSecurityProcessor.java

```diff
@@ -87,8 +84,8 @@ static boolean hasSandboxedNodes() {
         return SANDBOXED_NODES_COUNTER.get() > 0;
     }
 
-    /** Current security context. */
-    private final ThreadLocal<SecurityContext> curSecCtx = ThreadLocal.withInitial(this::localSecurityContext);
+    /** Current security context if differs from {@link #dfltSecCtx}. */
+    private final ThreadLocal<SecurityContext> curSecCtx = ThreadLocal.withInitial(() -> null);
```

Comment:
It seems that we can just use `new ThreadLocal()` here since the default value is null.